### PR TITLE
Add "report source" functionality #53 - updated

### PR DESCRIPTION
**Pull request by benboubekeur**
- Logged in users should be able to report a source
-   the "Report source" link should be in line with the "Mute source" link
-  Clicking the link brings you to a new page, before filing a report there's a required text area
-  Reports should be shown in the /admin/sources overview, ideally with a filter just like the "Show only inactive sources" filter
-  Admins should be able to read all reports for a source on that source's detail page

### CODE REVIEW

**Comment by brendt** [comment]
Thanks! You can target this towards master btw :) Could you take a look at the merge conflicts?

File: database/migrations/2020_11_06_160152_create_spams_table.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class CreateSpamsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('spams', function (Blueprint $table) {
```
Comment (brendt) [inline database/migrations/2020_11_06_160152_create_spams_table.php:16]: I'd call it `post_reports`, and the model `PostReport`

File: database/migrations/2020_11_13_182002_add_report_field_to_spams_table.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddReportFieldToSpamsTable extends Migration
```
Comment (brendt) [inline database/migrations/2020_11_13_182002_add_report_field_to_spams_table.php:7]: You can combine this with the original migration

**Review by brendt — CHANGES_REQUESTED** [review]
Already looking good! I've got a few remarks. There are also some code style things I'd do differently, but I'll fix those when I've merged this in master.

Since you started your branch, I updated Laravel; which means that the `spatie/blade-x` package was replaced by native Laravel blade components. The only things that should change is making sure all blade component files are `kebab-cased` and that components are prefixed with an `x`, eg `x-sort-link` instead of `sort-link`.

File: app/Domain/Post/Models/Post.php [hunk 3]
```diff
@@ -55,7 +55,7 @@ public function source(): BelongsTo
 
     public function votes(): HasMany
     {
-        return $this->hasMany(Vote::class);
+        return $this->hasMany(Spam::class);
```
[the remark this answers is not in the document]
Comment (benboubekeur) [reply app/Domain/Post/Models/Post.php:58]: Php storm was confused probably when refactoring !

**Comment by brendt** [comment]
Hey @benboubekeur I still think there's something wrong with your commit history: it lists all preview commits as well in the PR.

**Comment by benboubekeur** [comment]
So i should squash them ?

**Comment by brendt** [comment]
You can try that, have you rebased on master?

**Comment by benboubekeur** [comment]
Yes i did that :

- i will create a new branch locally with feature name

- Push that branch 

-  Create a new PR because it's messy now

**Comment by brendt** [comment]
Ok let's continue in the other PR

**Pull request closed without merging**